### ISSUE TITLE
lib/ipc/Makefile: mig command not found? (see last comment for cause)

### ISSUE THREAD

**Issue by cloudkitsch** [heimdal/heimdal#130]
```
make[2]: Entering directory '/home/erratic/heimdal-1.5.3/lib/ipc'
mig -header heim_ipc.h -user heim_ipcUser.c -sheader heim_ipcServer.h -server heim_ipcServer.c -I. ./heim_ipc.defs
make[2]: mig: Command not found
Makefile:1060: recipe for target 'heim_ipc.h' failed
make[2]: *** [heim_ipc.h] Error 127
make[2]: Leaving directory '/home/erratic/heimdal-1.5.3/lib/ipc'
Makefile:462: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/erratic/heimdal-1.5.3/lib'
Makefile:496: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
laptop heimdal-1.5.3 #
```

mig seems to be an OSX tool.. figures I wouldn't have it: 

```
laptop ~ # uname -a
Linux laptop 3.19.5-hardened #1 SMP PREEMPT Wed Apr 29 20:25:03 UTC 2015 x86_64 Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz GenuineIntel GNU/Linux
laptop ~ #
```

**Comment by cloudkitsch** [comment]
Same issue with the 1.5 branch git sources : 

```
make[3]: Leaving directory '/home/root/heimdal/lib/asn1'
make[2]: Leaving directory '/home/root/heimdal/lib/asn1'
Making all in sqlite
make[2]: Entering directory '/home/root/heimdal/lib/sqlite'
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc  -DHAVE_CONFIG_H -I. -I. -I../../include -I../../include  -I../../lib/roken -I../../lib/roken -I/usr/include/et  -D_LARGE_FILES= -Wall -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast -Wmissing-declarations -Wnested-externs  -g -O2 -MT sqlite3.lo -MD -MP -MF .deps/sqlite3.Tpo -c -o sqlite3.lo sqlite3.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I. -I../../include -I../../include -I../../lib/roken -I../../lib/roken -I/usr/include/et -D_LARGE_FILES= -Wall -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast -Wmissing-declarations -Wnested-externs -g -O2 -MT sqlite3.lo -MD -MP -MF .deps/sqlite3.Tpo -c sqlite3.c  -fPIC -DPIC -o .libs/sqlite3.o
sqlite3.c: In function ‘posixUnlock’:
sqlite3.c:26157:7: warning: variable ‘h’ set but not used [-Wunused-but-set-variable]
   int h;
       ^
sqlite3.c: In function ‘sqlite3RefillIndex’:
sqlite3.c:81469:7: warning: variable ‘regIdxKey’ set but not used [-Wunused-but-set-variable]
   int regIdxKey;                 /* Registers containing the index key */
       ^
sqlite3.c: In function ‘sqlite3CommitTransaction’:
sqlite3.c:82554:12: warning: variable ‘db’ set but not used [-Wunused-but-set-variable]
   sqlite3 *db;
            ^
sqlite3.c: In function ‘sqlite3RollbackTransaction’:
sqlite3.c:82574:12: warning: variable ‘db’ set but not used [-Wunused-but-set-variable]
   sqlite3 *db;
            ^
sqlite3.c: In function ‘trimFunc’:
sqlite3.c:344:38: warning: cast from function call of type ‘void *’ to non-matching type ‘long int’ [-Wbad-function-cast]
 # define SQLITE_PTR_TO_INT(X)  ((int)(__PTRDIFF_TYPE__)(X))
                                      ^
sqlite3.c:85158:13: note: in expansion of macro ‘SQLITE_PTR_TO_INT’
     flags = SQLITE_PTR_TO_INT(sqlite3_user_data(context));
             ^
sqlite3.c: In function ‘sqlite3Parser’:
sqlite3.c:108826:7: warning: variable ‘yyendofinput’ set but not used [-Wunused-but-set-variable]
   int yyendofinput;     /* True if we are at the end of input */
       ^
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I. -I../../include -I../../include -I../../lib/roken -I../../lib/roken -I/usr/include/et -D_LARGE_FILES= -Wall -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast -Wmissing-declarations -Wnested-externs -g -O2 -MT sqlite3.lo -MD -MP -MF .deps/sqlite3.Tpo -c sqlite3.c -o sqlite3.o >/dev/null 2>&1
mv -f .deps/sqlite3.Tpo .deps/sqlite3.Plo
/bin/sh ../../libtool  --tag=CC   --mode=link gcc  -Wall -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast -Wmissing-declarations -Wnested-externs  -g -O2   -o libheimsqlite.la -rpath /usr/heimdal/lib sqlite3.lo -pthread -lresolv -pthread
libtool: link: gcc -shared  -fPIC -DPIC  .libs/sqlite3.o   -lresolv  -g -O2 -pthread -pthread   -pthread -Wl,-soname -Wl,libheimsqlite.so.0 -o .libs/libheimsqlite.so.0.0.0
libtool: link: (cd ".libs" && rm -f "libheimsqlite.so.0" && ln -s "libheimsqlite.so.0.0.0" "libheimsqlite.so.0")
libtool: link: (cd ".libs" && rm -f "libheimsqlite.so" && ln -s "libheimsqlite.so.0.0.0" "libheimsqlite.so")
libtool: link: ar cru .libs/libheimsqlite.a  sqlite3.o
libtool: link: ranlib .libs/libheimsqlite.a
libtool: link: ( cd ".libs" && rm -f "libheimsqlite.la" && ln -s "../libheimsqlite.la" "libheimsqlite.la" )
make[2]: Leaving directory '/home/root/heimdal/lib/sqlite'
Making all in ipc
make[2]: Entering directory '/home/root/heimdal/lib/ipc'
mig -header heim_ipc.h -user heim_ipcUser.c -sheader heim_ipcServer.h -server heim_ipcServer.c -I. ./heim_ipc.defs
make[2]: mig: Command not found
Makefile:1414: recipe for target 'heim_ipc.h' failed
make[2]: *** [heim_ipc.h] Error 127
make[2]: Leaving directory '/home/root/heimdal/lib/ipc'
Makefile:547: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/root/heimdal/lib'
Makefile:588: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
laptop heimdal #

```

**Comment by cloudkitsch** [comment]
Just to follow up, it seems like this *might* be an issue with autotools or something on my system: 

```
laptop heimdal # libtool --version
libtool (GNU libtool) 2.4.6
Written by Gordon Matzigkeit, 1996

Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
laptop heimdal # autoreconf --version
autoreconf (GNU Autoconf) 2.69
Copyright (C) 2012 Free Software Foundation, Inc.
License GPLv3+/Autoconf: GNU GPL version 3 or later
<http://gnu.org/licenses/gpl.html>, <http://gnu.org/licenses/exceptions.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David J. MacKenzie and Akim Demaille.
laptop heimdal #
```

Someone else seems to not be having this problem:

```
03:36 -!- Irssi: #gentoo: Total of 1127 nicks [7 ops, 0 halfops, 55 voices, 1065 normal]
03:36 -!- Home page for #gentoo: http://www.gentoo.org
03:36 -!- Channel #gentoo created Sat Sep 15 03:15:36 2001
03:36 -!- Irssi: Join to #gentoo was synced in 2 secs
03:37 < erratic> hey was wondering if anyone here could verify that on their gentoo *linux* distribution that they have a /usr/bin/mig command?
03:37 < erratic> https://bugs.gentoo.org/show_bug.cgi?id=549136
03:37 <@xiaomiao> erratic: don't have it here
03:37 < erratic> I think you will have to if you've built heimdal
03:37 -!- kingkongmok [~kingkongm@121.8.124.244] has joined #gentoo
03:37 <@xiaomiao> which is not default
03:38 < erratic> right
03:38 < username234> erratic: I built heimdal earler today without issue.
03:40 < erratic> username234: can you verify that you have a command called mig and can you tell me what package provides it, I think equery can do it but let me check
03:40 < romiir> ok, so i guess emerge didnt add the services automagicly
03:40 < romiir> rebooting...
03:40 < romiir> i think all should be good after this
03:41 < romiir> i swear it said it added it to the default rc status though when we built it
03:41 -!- edannenbe [~ed@2a02:810a:1040:550:2dc7:db9c:fe29:127e] has quit [Quit: Leaving]
03:41 <@xiaomiao> romiir: emerge never does (except for a few like udev that are boot-critical)
03:42 < romiir> well
03:42 < erratic> username234: equery belongs `which mig`
03:42 < romiir> this is boot critical
03:42 < romiir> its vmware tools
03:42 < romiir> and it said it did
03:42 < romiir> and it did not
03:44 < romiir> finally after a few reboots and finangling
03:44 -!- ormaaj [~ormaaj@unaffiliated/ormaaj] has quit [Remote host closed the connection]
03:44 < romiir> its now working as it should
03:44 < username234> erratic: not part of that package
03:44 -!- ormaaj [~ormaaj@unaffiliated/ormaaj] has joined #gentoo
03:44 < erratic> username234: where does mig come from
03:44 < romiir> and not asking if i wish to pull the power cord out of our gentoo machine lol
03:44 < username234> erratic: don't know.
03:44 < erratic> but you have it?
03:45 < erratic> username234: can you tell me what equery says when you run this command: "equery belongs mig"
03:46 < username234> erratic: * Searching for mig ...
03:46 < username234> and then prompt
03:46 < erratic> what happens when you run "which mig"
03:46 < username234> erratic: the file does not exist.
03:47 < erratic> what version of heimdal do you have installed: "emerge -p heimdal"
03:47 < username234> erratic: there's only one version in my tree. 1.5.3-r2
03:47 < erratic> can you show me the output of "emerge -p heimdal"
03:48 -!- tscholak [~tscholak@174.115.92.130] has quit [Quit: ZZZzzz…]
03:48 < username234> erratic: did you read the github issue?  mig is an OS X tool.
03:49 < erratic> username234: I opened the github issue, I'm trying to find out why heimdal is not building on linux
03:49 < username234> erratic:  https://bpaste.net/show/3320b9550324
03:49 < erratic> my build is breaking when it tries to run the mig command which it shouldn't for obvious reasons
03:50 -!- jsubl2 [~jim@159-118-63-131.cpe.cableone.net] has quit [Quit: leaving]
03:51 -!- inkerman [~inkerman@d110-33-96-64.mas800.nsw.optusnet.com.au] has quit [Ping timeout: 256 seconds]
03:51 < erratic> alright well assuming you are telling the truth and that actually is linux then I guess there must be some issues with automake / autoconf / libtool on this setup that is messing up

```

**Comment by cloudkitsch** [comment]
With a little help from a guy on IRC, we determined that the cause is libdispatch, which I do in fact have installed on here because I was using it to try to get darling to work: 
```
04:32 -!- Irssi: Starting query in freenode with username234
04:32 <username234> just confirmed it.  If heimdal detects libdispatch it issues the mig commands.  The ebuild for libdispatch is not in Gentoo's main tree which is why  my build attempts didn't fail (do not have
                    libdispatch.so installed and cannot install libdispatch.so as I don't use overlays).   Easiest fix would probably be to disable heimdal's checking for libdispatch if the output of uname is anythong other
                    than 'Darwin' in configure.ac.
04:32 <username234> In any case, I have to go, so best of luck.
05:08 <erratic> you're absolutely right
05:08 <erratic> because I have libdispatch installed, was using it to try to get darling to work :)
05:08 <erratic> thank you
```

```
laptop portage # emerge --search libdispatch

[ Results for search key : libdispatch ]
Searching...

*  dev-libs/libdispatch
      Latest version available: 9999
      Latest version installed: 9999
      Size of files: 0 KiB
      Homepage:      http://nickhutchinson.me/libdispatch
      Description:   Linux port of Apple's open-source concurrency library
      License:       BSD

[ Applications found : 1 ]

laptop portage # uname -a
Linux laptop 3.19.5-hardened #1 SMP PREEMPT Wed Apr 29 20:25:03 UTC 2015 x86_64 Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz GenuineIntel GNU/Linux
laptop portage #

```

**Comment by nicowilliams** [comment]
Yes, there's a libdispatch for Linux that tricks Heimdal's `./configure` into thinking that you have GCD, because... you do.  But libdisptatch doesn't include an implementation of `mig`.

The thing to do seems to be: `AC_CHECK_PROGS(GCD_MIG, mig, [no])` in addition to checking for `dispatch_async_f()`.

**Comment by nicowilliams** [comment]
Fixed with 99b79d1.
